packages/grafana-ui/src/components/Input/Input.tsx

```diff
@@ -41,13 +53,25 @@ export const Input = forwardRef<HTMLInputElement, Props>((props, ref) => {
   const [prefixRef, prefixRect] = useMeasure<HTMLDivElement>();
   const [suffixRef, suffixRect] = useMeasure<HTMLDivElement>();
 
+  // Specifying a width on Input controls the external size of the input, which is taken up by the text input itself
+  // and suffix and prefix.
+  // However AutoSizeInput wants to pass in a width to control the size of the text input itself, but suffix/prefix
+  // eats up that space and makes the text clip
+  // To fix this, we have private context that tells Input when it's being used in an AutoSizeInput and increases
+  // the width by the size of the suffix/prefix
```

Comment:
Would maybe just add that this approach was chosen so that we don't pollute the `Input` public API with similar prop, as that may be sort of nagging question here.